docs/TEMPORARY_DIRECTORIES.md

```diff
@@ -0,0 +1,201 @@
+   mechanism that ensures all generated temporary files are subject to ageing
+   when the program that owns them abnormally dies, but not while it is still
+   running. This is a particularly interesting option when decompressing
+   tarballs that contain files with old modification/access times, as extracted
+   files are otherwise immediately candidates for deletion by the ageing
+   algorithm. The [`flock`](http://man7.org/linux/man-pages/man1/flock.1.html)
+   tool of the `util-linux` packages makes this concept available to shell
+   scripts. Note that `systemd-tmpfiles` only checks for BSD file locks on
+   directories, locks on other types of file nodes (including regular files)
+   are not considered.
+
+4. Keep the access time of all temporary files created current. In regular
+   intervals, use `utimensat()` or a related call to update the access time
+   ("atime") of all files that shall be kept around. Since the ageing algorithm
+   looks at the access time of files when deciding whether to delete them, it's
+   sufficient to update their access times in sufficiently frequent intervals to
+   ensure the files are not deleted. Since most applications (and tools such as
+   `ls`) primarily care for the modification time (rather than the access time)
+   using the access time for this purpose should be acceptable.
+
+5. Set the "sticky" bit on regular files. The ageing logic skips deletion of
+   all regular files that have the sticky bit (`chmod +t`) set. This is
+   honoured for regular files only however, and has no effect on directories as
+   the sticky bit has a different meaning for them.
+
+6. Don't use `/tmp/` or `/var/tmp/`, but use you own sub-directory under `/run/`
+   or `$XDG_RUNTIME_DIRECTORY` (the former if privileged, the latter if
+   unprivileged), or `/var/lib/` and `~/.config/` (similar, but with
+   persistency and suitable for larger data). The two temporary directories
+   `/tmp/` and `/var/tmp/` come with the implicit clean-up semantics described
+   above. When this is not desired, it's possible to create private per-package
+   runtime or state directories, and place all temporary files there. However,
+   do note that this means opting out of any kind of automatic clean-up, and it
+   is hence particularly essential that the program cleans up generated files
+   in these directories when they are no longer needed, in particular when the
+   program dies abnormally.
+
+5. Exclude your temporary files from clean-ups via a `tmpfiles.d/` drop-in. The
+   `x`/`X` line types may be used to exclude files matching the specified
+   globbing patterns from the ageing logic. If this is used, automatic clean-up
+   is not done for matching files and directory, and much like with the
+   previous option it's hence essential that the program generating these
+   temporary files carefully removes the temporary files it creates again, and
+   in particular so if it dies abnormally.
```

Comment:
I don't think "abnormally" is the right term here. It usually means being killed by a signal. Maybe just drop the whole part starting with "again, and in particular so".

Please add a hint here that this tmpfiles configuration may be stored under /run too.

Reply:
I changed "abnormally" to "unexpectedly" throughout the whole document. I think that should address your point? (At least I think it's a much more approriate term than "abnormal" for this, indeed)